pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,383 @@
+	"github.com/spf13/cobra"
+)
+
+type options struct {
+	selectedClusterId             string
+	clusterManagementApiUrl       string
+	accessToken                   string
+	clusterList                   []clustersmgmtv1.Cluster
+	selectedCluster               clustersmgmtv1.Cluster
+	existingMachinePoolList       []clustersmgmtv1.MachinePool
+	selectedClusterMachinePool    clustersmgmtv1.MachinePool
+	requestedMachinePoolNodeCount int
+	accessKafkasViaPrivateNetwork bool
+	pageNumber                    int
+	pageSize                      int
+
+	f *factory.Factory
+}
+
+// list of consts should come from KFM
+const (
+	machinePoolId           = "kafka-standard"
+	machinePoolTaintKey     = "bf2.org/kafkaInstanceProfileType"
+	machinePoolTaintEffect  = "NoExecute"
+	machinePoolTaintValue   = "standard"
+	machinePoolInstanceType = "r5.xlarge"
+	machinePoolLabelKey     = "bf2.org/kafkaInstanceProfileType"
+	machinePoolLabelValue   = "standard"
+	clusterReadyState       = "ready"
+	fleetshardAddonId       = "kas-fleetshard-operator"
+	strimziAddonId          = "managed-kafka"
+	fleetshardAddonIdQE     = "kas-fleetshard-operator-qe"
+	strimziAddonIdQE        = "managed-kafka-qe"
+)
+
+func NewRegisterClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "register-cluster",
+		Short:   f.Localizer.MustLocalize("dedicated.registerCluster.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("dedicated.registerCluster.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("dedicated.registerCluster.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runRegisterClusterCmd(opts)
+		},
+	}
+
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	flags.StringVar(&opts.clusterManagementApiUrl, "cluster-mgmt-api-url", "", f.Localizer.MustLocalize("dedicated.registerCluster.flag.clusterMgmtApiUrl.description"))
+	flags.StringVar(&opts.accessToken, "access-token", "", f.Localizer.MustLocalize("dedicated.registercluster.flag.accessToken.description"))
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("dedicated.registerCluster.flag.clusterId.description"))
+	flags.IntVar(&opts.pageNumber, "page-number", int(cmdutil.ConvertPageValueToInt32(build.DefaultPageNumber)), f.Localizer.MustLocalize("dedicated.registerCluster.flag.pageNumber.description"))
+	flags.IntVar(&opts.pageSize, "page-size", 100, f.Localizer.MustLocalize("dedicated.registerCluster.flag.pageSize.description"))
+
+	return cmd
+}
+
+func runRegisterClusterCmd(opts *options) (err error) {
+	opts.pageNumber = int(cmdutil.ConvertPageValueToInt32(build.DefaultPageNumber))
+	err = getPaginatedClusterList(opts)
+	if err != nil {
+		return err
+	}
+	if len(opts.clusterList) == 0 {
+		return opts.f.Localizer.MustLocalizeError("dedicated.registerCluster.run.noClusterFound")
+	}
```

Comment:
Let's consider asking the under to try another page or adjust the page size. It could that the cluster they are looking is on the next page.

Reply:
great shout 👍